src/magic_folder/uploader.py

```diff
@@ -95,7 +96,10 @@ def store_local_snapshot(self, path):
             # Query the db to check if there is an existing local
             # snapshot for the file being added.
             # If so, we use that as the parent.
-            mangled_name = magicpath.mangle_path(path)
+
+            mangled_name = mangle_path_segments(
+                path.asTextMode("utf8").segmentsFrom(self._magic_dir.asTextMode("utf8"))
```

Comment:
I think we should be converting these paths to unicode closer to the edge of the system.